resource_detectors/test/opentelemetry/detectors/deployment_test.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+describe OpenTelemetry::Resource::Detectors::Deployment do
+  let(:detector) { OpenTelemetry::Resource::Detectors::Deployment }
+
+  describe '.detect' do
+    let(:detected_resource) { detector.detect }
+    let(:detected_resource_attributes) { detected_resource.attribute_enumerator.to_h }
+    let(:expected_resource_attributes) { {} }
+
+    it 'returns an empty resource' do
+      _(detected_resource).must_be_instance_of(OpenTelemetry::SDK::Resources::Resource)
+      _(detected_resource_attributes).must_equal(expected_resource_attributes)
+    end
+
+    describe 'when in a rack environment' do
```

Comment:
Please consider adding a test case for when the environment variable is `nil` and the `Rails.env` is undefined

```
irb(main):003:0> ENV['RACK_ENV']
=> nil
````

Reply:
I meant for the first test "returns an empty resource" to check this case. Is it the case? or should I refactor it to match your suggestion?

Reply:
That is fair, although it would mean when executing the test the environment must not have RACK_ENV set otherwise it would result in the test failing.

Reply:
good point, added clearing the env before this test:
```rb
      with_env({}) do
        _(detected_resource).must_be_instance_of(OpenTelemetry::SDK::Resources::Resource)
        _(detected_resource_attributes).must_equal(expected_resource_attributes)
      end
```